applications/mne_analyze/plugins/coregistration/coregistration.cpp

```diff
@@ -0,0 +1,756 @@
+                                                matHsp,
+                                                transInit,
+                                                vecTake,
+                                                matHspClean,
+                                                fMaxDist)) {
+        qWarning() << "Discard outliers was not succesfull.";
+    }
+    int iNDiscarded = vecWeightsICP.size() - vecTake.size();
+    m_pCoregSettingsView->setOmittedPoints(iNDiscarded);
+
+    VectorXf vecWeightsICPClean(vecTake.size());
+    for(int i = 0; i < vecTake.size(); ++i) {
+        vecWeightsICPClean(i) = vecWeightsICP(vecTake(i));
+    }
+
+    // icp
+    RTPROCESSINGLIB::performIcp(mneSurfacePoints,
+                                matHspClean,
+                                transInit,
+                                fRMSE,
+                                bScale,
+                                iMaxIter,
+                                fTol,
+                                vecWeightsICPClean);
+
+    FiffCoordTrans transHeadMri = transInit;
+
+    // update GUI
+    Vector3f vecRot;
+    Vector3f vecScale;
+    Vector3f vecTrans;
+    getParamFromTrans(transHeadMri.trans,vecRot,vecTrans,vecScale);
+    m_pCoregSettingsView->setTransParams(vecTrans,vecRot,vecScale);
+    m_pCoregSettingsView->setRMSE(fRMSE);
+
+    return transHeadMri;
+}
+
+//=============================================================================================================
+
+void CoRegistration::createNewTrans()
+{
+    m_transHeadMri = m_Future.result();
+
+    // update GUI
+    Vector3f vecRot;
+    Vector3f vecScale;
+    Vector3f vecTrans;
+    getParamFromTrans(m_transHeadMri.trans,vecRot,vecTrans,vecScale);
+    m_pCoregSettingsView->setTransParams(vecTrans,vecRot,vecScale);
+
+    // send event
+    QVariant data = QVariant::fromValue(m_transHeadMri);
+    m_pCommu->publishEvent(EVENT_TYPE::NEW_TRANS_AVAILABE, data);
+
+    triggerLoadingEnd("Performing ICP ...");
+}
+
+//=============================================================================================================
+
+void CoRegistration::onUpdateTrans()
+{
+    Vector3f vecRot;
+    Vector3f vecScale;
+    Vector3f vecTrans;
+    Matrix4f matTrans;
+
+    // update transformation
+    m_pCoregSettingsView->getTransParams(vecRot,vecTrans,vecScale);
+    getTransFromParam(matTrans,vecRot,vecTrans,vecScale);
+
+    // send event
+    m_transHeadMri.trans = matTrans;
+    QVariant data = QVariant::fromValue(m_transHeadMri);
+    m_pCommu->publishEvent(EVENT_TYPE::NEW_TRANS_AVAILABE, data);
+
+    return;
+}
+
+//=============================================================================================================
+
+void CoRegistration::getParamFromTrans(const Matrix4f& matTrans,
+                                       Vector3f& vecRot,
+                                       Vector3f& vecTrans,
+                                       Vector3f& vecScale)
+/**
```

Comment:
this documentation should go into the header file where the function is declared.